### DISCUSSION TITLE
Stacking mechanic feedback

### DISCUSSION THREAD

**Discussion by Kulik242** [FakeFishGames/Barotrauma#5041] [Ideas and suggestions]
Stacking small items such as revolver rounds and shotgun shells is great.

Players can now carry less item types due to toolbox reduction. I’m not necessarily against this as this forces more cooperative approach (deciding on who takes what), but it makes the singleplayer a lot harder.

I don’t like the stacking of larger items in player’s inventories because:
1.	It makes no sense – You can carry 80 pieces of ore in your inventory but only 10 if they are of different type. The graphical representation of those “medium” items also doesn’t suggest that they should be stackable on players.
2.	Stacking oxygen tanks trivializes the time pressure aspect of diving missions which was a very integral part of leaving the sub. Same goes with batteries for scooters, sonars and flashlights.
3.	Fact that you can carry 8 tanks in one slot, but carrying 1 full, 1 empty and 1 half-empty tank takes 3 slots, i.e. almost third of your inventory is very counterintuitive.
4.	Traitors/griefers can incapacitate the sub by easily stealing large quantities of important assets (fuel rods, oxygen tanks, welding fuel tanks, medical items).

My proposed solution:
1. Containers and cabinets should allow stacking, player inventories should allow only the stacking of small items such as ammo or medicals.
2. Leave the reduced number of slots in toolbelts but add a floating (or veeeery slowly sinking) type of a container. Or make scooter have storage capacity, but make it so that the scooter itself can’t be stored in inventory and is floating after un-equipping.
3.	Make fuel rods big (railgun shell sized) with a specialized fuel storage cabinet where fuel rods can stack.
4.	No idea how to prevent griefers from easily stealing your 8xDeusizine stack. :-)

Additional suggestion not related to stacking but relevant to inventory management: 
Limit on the number of two handed weapons that players can carry. I'd be in favour of one on the back in addition to one wielded. Perhaps two on back + one wielded, but I like more the idea of having a dedicated grenadier, SMG guy, shotgun guy, etc.
In addition, these weapons should be graphically represented on the ragdoll in a way as toolbelts are. So clowns (and griefers) couldn't pull a grenade launcher in your face from their magical pocket.

**Comment by Xeavas** [comment]
I would like to comment about ores being not stackable would create some problems, especially about eva mining, you would need to implement a reliable way of cargo transportation while out mining, for example, cargo drones that follow you or crates that float in place. or a eva backpack.

**Reply by Kulik242** [reply]
Hence the scooter with cargo capacity or floating containers.

**Reply by Xeavas** [reply]
yeah, a crate that stays in place like glowsticks would be good

**Comment by ubertpendragon** [comment]
>     2. Stacking oxygen tanks trivializes the time pressure aspect of diving missions which was a very integral part of leaving the sub. Same goes with batteries for scooters, sonars and flashlights.
> 
>     3. Fact that you can carry 8 tanks in one slot, but carrying 1 full, 1 empty and 1 half-empty tank takes 3 slots, i.e. almost third of your inventory is very counterintuitive.

This really bothers me; not just carrying near-infinite amounts of oxygen in limited spaces (wrecks have their own oxygen tanks to boot), but the fact that it expands once you actually start using the bottles.

Tanks using partials causes conflict with the stacking mechanic. Oxygen Tanks should not be stackable. If this causes balance issues, you can still increase the duration of a tank, but it's a good example of stacking going _too far_


Also, **Storage Containers still exist**, still have 12 slots, and can still fit within cabinets. 
With resources stacking up to 8 times, a large steel cabinet can hold up to 30x12x8 inventory slots  for a whopping 2880 item capacity

Storage containers are objectively better crates; and should not be in the game now that we have stacking implemented

**Reply by Sviat** [reply]
> Storage containers are objectively better crates; and should not be in the game now that we have stacking implemented

Sure they are bad now, they have to be stackable as well!

**Comment by Ek-Videogames** [comment]
(Vanilla) submarines only have at most a single large steel cabinet for storing almost everything on the ship, making the use of inventory compression still very much required in those situations. - While on submarines which are designed to handle items without storage containers become way too much compression of space if you **do** decide to use storage containers instead. 
_"Unnecessary if you have a solution already, but impossible to live without if you don't"_
This is another situation like toolboxes where you should **give people a better solution to the problem, instead of removing the current solution immedeately.**

Storage containers just feel like a worse version of metal-crate, and their inclusion kind of stopped making sense the moment they became not-toolboxes. Right now its like we have a bandaid on top of a bandaid on top of the actual issue, which is inventory management.

> " Giving us stacking but then nerfing container sizes to oblivion does not work because holding 1 each of 8 different types of medicines for instance will not compare to holding a full 8-stack of one medicine. From my point of view I was hoping that item stacking was going to be used in order to allow items that are 'Smaller than small' IE items that would make sense to be able to stack - Raw resources / chems, revolver bullets, flares / glowsticks, maybe battery cells. (To be honest stackable oxygen tanks basically 'solves' the issue of oxygen tanks lasting so short, but Ideally at this point it would make sense just to have 'bigger' tanks that instead don't stack, but hold more air.) Harpoon gun, Revolver, and syringe gun being single slot only is dumb because the same reason, you can't "Mixed load" especially in the case of harpoon gun ammo and syringe gun ammo types. "

**Reply by ubertpendragon** [reply]
I think it's worth writing a summary of storage solutions that are currently in the game, so here's a quick list of what's wrong in terms of inventory compression:

- **Storage containers** hold 12 items, are 1-handed, and go inside cabinets. This means that you can carry 24 slots with you (same as a steel crate), but also "compress" so hard you can fit up to 2880 items in a large cabinet. This is pretty broken, and the mere existence of storage containers isn't justified after stacking, a removal is in order. After all, why not use a crate as spare storage? Because...

- **Crates** are supposed to be the default, mobile item transport solution, but have a few issues. As EK said, _vanilla submarines have terrible storage space management_. The Berilia, Orca, R29, Kastrull, Typhon 1/2 and worst of all the Remora don't even have crate shelves near their fabricator, mostly relying on a single large cabinet to store items. Your only option is leave crates on the ground, flying across the room when Hammerheads swim by. This isn't crates being too weak; having as many 20 slot crates on the ground as you want is **infinite storage space**. But it does look silly and is inconvenient. The true negative aspect of crates is how cumbersome they are in EVA sorties, as always they sink to the bottom. Adding a one-slot crate holder to all of those subs would do wonders, as well adding as a crate type that floats for diving teams.

* **Toolbelts** are good... too good? I don't think anybody would complain about having 6 extra inventory slots, but where is the **opportunity cost**? Having a toolbelt is definitely better than having no toolbelt, 16 slots is better than 10... Where are the other toolbelts types for the toolbelt slot? Other "hidden" aspect of toolbelts is that, unlike Storage Containers, they can fit inside crates. **Toolbelt inventory compression** means you can multiply crate and cabinet slots by 6 slots. The only thing that detracts most players from filling every crate with toolbelts is how inconvenient moving toolbelts is, one at a time, and having to drop your own toolbelt... Toolbelts probably shouldn't fit inside crates; it's too easy to go salvage a wreck and dump items inside your 24x6 slot crate, and thankfully players won't bother using Toolbelt compression in cabinets because it is a pain in the ass to hot-swap, unlike Storage containers.

* **Duffle Bags** are... one-handed crates. I don't have to explain how powerful they are for looting purposes. It's not really justified; 20 slots while the inventory has 10. Why? Because somehow they should also fit the dead person's radio and clothes. Which quite frankly are "free" money, dead bodies print clothes + radio + battery worth of Mks. So, please, nerf duffles.

* **Corpses**. Yes, corpses are a storage solution. Whenever you go looting a wreck, the optimal technique is finding a dead body and loading its 10 inventory slots (16 if toolbelt)! With as much crap as you can find, put a diving suit on it, grab it, and scoot all the way back to the sub. Corpses spawn with at most 4-5 slots worth of inventory space; they really don't need 10 inventory slots, a toolbelt slot, and a diving suit slot. Once again, asking for a nerf

I think that covers it. 
tl;dr: 

Remove Storage containers
Crate shelves for vanilla subs. 1-slot crate shelf variant. Floating crates
Toolbelts shouldn't go inside crates, add more alternative toolbelts
Reduce duffle bag slots from 20 to 10, make them two-handed, no more free radios
Reduce wreck corpse slots from 10 to 5, no toolbelt/suit slot

Also: 
> Giving us stacking but then nerfing container sizes to oblivion does not work because holding 1 each of 8 different types of medicines for instance will not compare to holding a full 8-stack of one medicine

I feel like it's a consequence of having many, many resources, and some of those being entirely useless, clogging up the storage. Think about Calcium, Zinc, Tin, Silicon...

**Comment by Krzeszny** [comment]
I'll add my idea for a medical/syringe "bandoleer" that functions similarly to a doctor's outfit, except it can only transport syringes. 
Perhaps it could be capped to a max amount of individual syringes instead of slots...

TBH, the stacking system needs a rework.

**Comment by HeatedWafflez** [comment]
_"Stacking oxygen tanks trivializes the time pressure aspect of diving missions which was a very integral part of leaving the sub. Same goes with batteries for scooters, sonars and flashlights."_

Yes! I absolutely agree with you and have brought this up myself. I strongly believe that exiting the submarine should be an event that every player has to thoroughly plan and prepare for beforehand. Being able to stack so many oxygen tanks effectively diminishes the danger of running out of oxygen, especially given how long each tank lasts. 

I know that some portions of the game might come across as too difficult to the developers. This might be true sometimes but in this specific instance this is crucial because it captures the very essence of this game and the setting it takes place in. The game takes place in a hostile underwater world of dangerous creatures, anomalies, and seemingly infinite darkness. It's not supposed to be easy.
